roles/provision_virtual_esxi/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Check General Mandatory Variables Are Defined
+  ansible.builtin.assert:
+    that:
+      - provision_virtual_esxi_username is defined
+      - provision_virtual_esxi_password is defined
+      - provision_virtual_esxi_hostname is defined
+      - provision_virtual_esxi_vm_names is defined
+    quiet: true
+    fail_msg: Variable must be set when using this role.
+
+- name: Check If Hosts Exist
+  community.vmware.vmware_guest_info:
+    hostname: "{{ provision_virtual_esxi_hostname }}"
+    username: "{{ provision_virtual_esxi_username }}"
+    password: "{{ provision_virtual_esxi_password }}"
+    datacenter: "{{ provision_virtual_esxi_datacenter | default('') }}"
+    validate_certs: "{{ provision_virtual_esxi_validate_certs | default(omit) }}"
+    name: "{{ _esxi_name }}"
+  failed_when: false
+  register: _esxi_host_check
+  loop: "{{ provision_virtual_esxi_vm_names }}"
+  loop_control:
+    loop_var: _esxi_name
+
+- name: Include New ESXi Hosts Tasks
```

Comment:
If the `deploy_new_esxi_hosts.yml` would you need to do this check? I think it shouldn't be needed.

Reply:
Ah, sorry I mistyped the question. I was asking whether it's really needed to have the check in the task. If the `deploy_new_esxi_hosts.yml` is idempotent, then no checks are needed and it can still go through all the steps. And to achieve and also... if you update for example memory in variables, the hosts will be updated, so the role will be idempotent, this way I think it's not.

Reply:
ok i see. yea right now its not. im going to resolve this thread because the other thread we have is related and will influence this check